File: javascript/24.js
let limit=1000000

function fact(n){
    if(n<2) return 1    
    return n*(fact(n-1))
}

let number = []
let digits = [0,1,2,3,4,5,6,7,8,9]
let count=0
for(let i = 9;i>=0;i--){    
    let start = digits.length
    
    let checker = start*fact(i)+count
    while(checker>=limit ){
        start--
        checker = start*fact(i)+count
    }
    count+=(start)*fact(i)
    number.push(digits.splice(start,1))
    
}

console.log(number.join(""));

File: site/python/24.py
def fact(n):
    if n<2: return 1
    return n*fact(n-1)

limit=1000000
digits = list(range(0,10))
number =[]
limitCounter=0

for i in range(9,-1,-1):
    start = len(digits)
    while start*fact(i)+limitCounter>=limit:
        start-=1
    limitCounter+=start*fact(i)
    number.append(digits[start])
    digits.pop(start)
print( int("".join(map(str, number))))